tests/xtimer_mutex_lock_timeout/main.c

```diff
@@ -42,9 +49,38 @@ static const shell_command_t shell_commands[] = {
       cmd_test_xtimer_mutex_lock_timeout_long_unlocked, },
     { "mutex_timeout_long_locked", "locked mutex with long timeout",
       cmd_test_xtimer_mutex_lock_timeout_long_locked, },
+    { "mutex_timeout_long_locked_low",
+      "locked mutex from lower prio thread function with long timeout",
+      cmd_test_xtimer_mutex_lock_timeout_low_prio_thread, },
     { NULL, NULL, NULL }
 };
 
+/**
+ * @brief   thread function and stack for
+ *          cmd_test_xtimer_mutex_lock_timeout_low_prio_thread
+ */
+
+char t_stack[THREAD_STACKSIZE_MAIN];
+
+void *thread_low_prio_test(void *arg)
+{
+    mutex_t *test_mutex = (mutex_t *)arg;
+    msg_t msg;
+
+    puts("THREAD low prio: start");
+
+    mutex_lock(test_mutex);
+    thread_wakeup(main_thread_pid);
+    while (msg_try_receive(&msg) == -1) {}
```

Comment:
first: why not just blocking call of `msg_receive` instead of this loop? second: why use msg send/recv here at all? shouldn't the `thread_sleep() + thread_wakeup()` be enough to lock the mutex in the low prio thread before the main thread continues?

Reply:
after `thread wakeup ` the main thread runs anyways. I don't need that line it is only to better understand what is happening (the thread waits but is still runnable)
The thread should only check receive once.
I can remove it if it is more confusing than helpful.

Reply:
removed it